x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/autoscaling/MlAutoscalingResourceTracker.java

```diff
@@ -134,12 +145,12 @@ static void getMemoryAndProcessors(
             ? 0L
```

Comment:
I think that heterogeneous nodes are a gray area in autoscaling. I think that we should support heterogenous nodes, but I think that we currently do not support heterogenous nodes. In other parts of the autoscaling codebase you can find references to this. I'm going to add this to the meta issue because I don't think its relevant to this PR but is something we should probably tackle.